macOS/DuckDuckGo/Freemium/DBP/FreemiumDBPFeature.swift

```diff
@@ -134,55 +146,142 @@ final class DefaultFreemiumDBPFeature: FreemiumDBPFeature {
             .sink { [weak self] _ in
                 guard let self = self else { return }
 
-                let featureAvailable = self.isAvailable
-                Logger.freemiumDBP.debug("[Freemium DBP] Subscription Updated. Feature Availability = \(featureAvailable)")
-
-                self.isAvailableSubject.send(featureAvailable)
+                Logger.freemiumDBP.debug("[Freemium DBP] Subscription Updated. Feature Availability = \(self.isAvailable)")
+                self.isAvailableSubject.send(self.isAvailable)
             }
             .store(in: &cancellables)
+
+        // Subscribe to available product updates for App Store Environment
+        if subscriptionManager.currentEnvironment.purchasePlatform == .appStore {
+            subscriptionManager.canPurchasePublisher
+                .sink { [weak self] canPurchase in
+                    guard let self = self else { return }
+
+                    // Calculate availability manually for App Store to include purchase capability
+                    // This provides more granular control over the availability calculation
+                    // compared to the general isAvailable property
+                    let featureEnabled = isEligibleFeatureFlagEnabled
+                    let notCurrentUser = isEligibleNotCurrentUser
+                    let storeFrontIsUSA = isEligibleAppStorefrontIsUSA
+                    let available = featureEnabled && notCurrentUser && storeFrontIsUSA && canPurchase
+                    Logger.freemiumDBP.debug("[Freemium DBP] Subscription Updated. Feature Availability = \(available)")
+
+                    self.isAvailableSubject.send(available)
+                }
+                .store(in: &cancellables)
+        }
     }
 }
 
+/// Private extension containing eligibility logic and helper methods.
 private extension DefaultFreemiumDBPFeature {
 
-    /// Returns true IFF:
+    // MARK: - Freemium Eligibility Logic
+
+    /// Determines overall eligibility by combining feature flag, auth status, storefront, and purchase capability.
+    var isEligible: Bool {
+        isEligibleFeatureFlagEnabled && isEligibleNotCurrentUser && isEligibleAppStorefrontIsUSA && isEligibleCanPurchase
```

Comment:
Naming of the main var and the compound ones is not clear to me. I suggest:
- keep the main one as `isEligible` or make if fully descriptive `isEligibleForFreemium...`
- the subsequent condition vars have their name starting with `isEligible...` but to me they do not guarantee eligibility, maybe to be clearer about that we can rename them with a pattern like `isEligibleFeatureFlagEnabled` -> `canBeEligibleDueToFeatureFlagEnabled` or maybe drop the whole `isEligible` prefix and be clear about what condition is checked e.g. `isFeatureFlagEnabled`, WDYT?

Reply:
Thanks @miasma13, agree. Naming updated.